### PR TITLE
fix parsing error caused by single quote transformation

### PR DESCRIPTION
**Pull request by v01dstar**
PTAL @coocood @shenli 

Fix string unquote bug:

SQL like: SELECT 'test"test'; can be executed correctly.

Originally, 'test"test' will be converted into "test"test" which will cause the parsing error.

### CODE REVIEW

File: util/stringutil/string_util.go [hunk 1]
```diff
@@ -60,3 +62,181 @@ func Reverse(s string) string {
 	}
 	return string(r)
 }
+
+func UnquoteChar(s string, quote byte) (value rune, multibyte bool, tail string, err error) {
```
Comment (shenli) [inline util/stringutil/string_util.go]: Please add a few unit test cases for this function, and also a few comments.

File: util/stringutil/string_util.go [hunk 2]
```diff
@@ -60,3 +62,181 @@ func Reverse(s string) string {
...
+		} else {
+			n := utf8.EncodeRune(runeTmp[:], c)
+			buf = append(buf, runeTmp[:n]...)
+		}
+	}
+	return string(buf), nil
+}
+
+// contains reports whether the string contains the byte c.
+func contains(s string, c byte) bool {
```
Comment (shenli) [inline util/stringutil/string_util.go]: You can use this function: https://golang.org/pkg/strings/#IndexByte

**Comment by v01dstar** [comment]
PTAL @shenli @coocood

**Comment by shenli** [comment]
Well done！
LGTM

**Comment by coocood** [comment]
LGTM

File: util/stringutil/string_util.go [hunk 3]
```diff
@@ -60,3 +62,185 @@ func Reverse(s string) string {
...
+// If set to a single quote, it permits the sequence \' and disallows unescaped '.
+// If set to a double quote, it permits \" and disallows unescaped ".
+// If set to zero, it does not permit either escape and allows both quote characters to appear unescaped.
+// Diffrent with strconv.UnquoteChar, it permits unnecessary backslash.
+func UnquoteChar(s string, quote byte) (value rune, multibyte bool, tail string, err error) {
+	// easy cases
+	switch c := s[0]; {
+	case c == quote && (quote == '\'' || quote == '"'):
+		err = strconv.ErrSyntax
+		return
```
Comment (zimulala) [inline util/stringutil/string_util.go]: `return` -> `return errors.Trace(err)` ?

**Comment by zimulala** [comment]
LGTM

**Pull request merged**